weave/trace_server/migrations/007_stats_views.up.sql

```diff
@@ -0,0 +1,138 @@
+    created_at SimpleAggregateFunction(min, DateTime64(3)),
+    updated_at SimpleAggregateFunction(max, DateTime64(3))
+) ENGINE = AggregatingMergeTree()
+ORDER BY (project_id, digest, chunk_index);
+
+CREATE MATERIALIZED VIEW files_stats_view
+TO weave_trace_db.files_stats
+AS
+SELECT
+    project_id,
+    digest,
+    chunk_index,
+    anySimpleState(n_chunks) as n_chunks,
+    anySimpleState(name) as name,
+    anySimpleState(length(val_bytes)) AS size_bytes,
+    minSimpleState(created_at) AS created_at,
+    maxSimpleState(created_at) AS updated_at
+FROM weave_trace_db.files
+GROUP BY
+    project_id,
+    digest,
+    chunk_index;
+
+CREATE TABLE table_rows_stats
+(
+    project_id String,
+    digest String,
+    refs SimpleAggregateFunction(any, Array(String)),
+    size_bytes SimpleAggregateFunction(any, UInt64),
+    created_at SimpleAggregateFunction(min, DateTime64(3)),
+    updated_at SimpleAggregateFunction(max, DateTime64(3))
+) ENGINE = AggregatingMergeTree()
+ORDER BY (project_id, digest);
+
+CREATE MATERIALIZED VIEW table_rows_stats_view
+TO weave_trace_db.table_rows_stats
+AS
+SELECT
+    project_id,
+    digest,
+    anySimpleState(refs) as refs,
+    anySimpleState(length(val_dump)) AS size_bytes,
+    minSimpleState(created_at) AS created_at,
+    maxSimpleState(created_at) AS updated_at
+FROM weave_trace_db.table_rows
+GROUP BY
+    project_id, digest;
+
+
+CREATE TABLE object_versions_stats
+(
+    project_id String,
+    kind Enum('op', 'object'),
+    object_id String,
+    digest String,
+    base_object_class SimpleAggregateFunction(any, Nullable(String)),
+    refs SimpleAggregateFunction(any, Array(String)),
+    size_bytes SimpleAggregateFunction(any, UInt64),
+    created_at SimpleAggregateFunction(min, DateTime64(3)),
+    updated_at SimpleAggregateFunction(max, DateTime64(3))
+) ENGINE = AggregatingMergeTree()
+ORDER BY (project_id, kind, object_id, digest);
+
+CREATE MATERIALIZED VIEW object_versions_stats_view
+TO weave_trace_db.object_versions_stats
+AS
+SELECT
+    project_id,
+    kind,
+    object_id,
+    digest,
+    anySimpleState(base_object_class) AS base_object_class,
+    anySimpleState(refs) AS refs,
+    anySimpleState(length(val_dump)) AS size_bytes,
+    minSimpleState(created_at) AS created_at,
+    maxSimpleState(created_at) AS updated_at
+FROM weave_trace_db.object_versions
+GROUP BY
+    project_id, kind, object_id, digest;
+
+CREATE TABLE calls_merged_stats
+(
+    project_id String,
+    id String,
+    trace_id SimpleAggregateFunction(any, Nullable(String)),
+    parent_id SimpleAggregateFunction(any, Nullable(String)),
+    op_name SimpleAggregateFunction(any, Nullable(String)),
+    started_at SimpleAggregateFunction(any, Nullable(DateTime64(3))),
+    attributes_size_bytes SimpleAggregateFunction(any, Nullable(UInt64)),
+    inputs_size_bytes SimpleAggregateFunction(any, Nullable(UInt64)),
+    input_refs SimpleAggregateFunction(groupArrayArray, Array(String)),
```

Comment:
not sure if you wanted to trim down the fields, but you could probably drop many of these if we just want the size stats.

Reply:
felt easier to include them..they might come in handy for quick analysis? we can always drop them later. easier to drop than include later as backfills are hard.